Test.cs

```diff
@@ -12,7 +14,12 @@ public static long calculate(long number)
                 throw new Exception("Number must be non-negative.");
             }
 
-            return number == 0 ? 1 : number * Calculate(number + 1);
```

Comment:
The recursive call in the `Calculate` method is incorrect. It should be `number * Calculate(number - 1)` instead of `number * Calculate(number + 1)`.